File: Max_Elemet/Max.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 16/07/2023
    Program : Accept N numbers from user and return the largest number.
              Input : N : 6
                      Elements : 85 66 3 66 93 88
              Output : 93 
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class max
{
    //Characterstics of Class
    isize : number;
    arr : number[];

    //Constructor of Class
    constructor()
    {
        this.isize = 0;
        this.arr = [];
    }

    //Behaviour of Class To Take input from user
    Accept()
    {
        this.isize = 6;
        this.arr = [85,66,3,66,88,93];
    }

    //Behaviour of Class To Find Out Max Number 
    Maximum()
    {
        var i : number = 0;
        var imax : number = 0;

        //Logic
        imax = this.arr[0];
        for(i = 0;i < this.isize;i++)
        {
            if(this.arr[i] > imax) 
            {
                imax = this.arr[i];
            }
        }

        return imax;
    }
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
function main()
{
    var iret : number = 0;

    console.log("Ganapati Bappa Morya.....");

    //Creating object of Max Class
    var mobj = new max();

    //Call To Behaviours of Class
    mobj.Accept()
    iret = mobj.Maximum();
    console.log("Maximum Number from Entered N Numbers is : %d",iret);
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Call To Main function
main();

File: Max_Elemet/Max.cpp
/*
    Author : Pakshal Shashikant Jain 
    Date : 16/07/2023
    Program : Accept N numbers from user and return the largest number.
              Input : N : 6
                      Elements : 85 66 3 66 93 88
              Output : 93 
*/

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraires
#include<iostream>
using namespace std;

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Max
{
    //Characteristics of Class
    public :
        int isize;
        int *arr;
        int i;

    //Constructor of Class
    public :
        Max()
        {
            this->i = 0;
            this->arr = NULL;
            this->isize = 0;
        }

    //Behaviour of Class To Take Input from user
    void Accept()
    {
        //Taking Input From user
        cout<<"Enter Number of Elements You Want in array\n";
        cin>>this->isize;

        //Handeling Conditions if isize is less than 0
        if(this->isize < 0) 
        {
            this->isize = -this->isize;
        }    

        //Allocating memory To array 
        this->arr = (int *)malloc(this->isize * sizeof(int));

        //Inserting Element in array 
        cout<<"Enter Elements in array \n";
        for(this->i = 0;this->i < this->isize;this->i++)
        {
            cout<<"Enter Element Number : "<<i + 1<<"\n";
            cin>>this->arr[this->i];
        }
    }

    //Behaviour of Class To Find max Element from entered N Numbers
    int max()
    {
        int max = 0;

        //Logic
        max = this->arr[0];
        for(this->i = 0;this->i < this->isize;this->i++)
        {
            if(this->arr[i] > max)
            {
                max = this->arr[i];
            }
        }

        return max;
    }
};

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
int main() 
{
    int iret = 0;

    printf("Ganapati Bappa Morya...\n");

    //Creating object of Max Class
    Max obj;

    //Call TO Behaviours of Class
    obj.Accept();
    iret = obj.max();   

    //Printing Max Element From Entered N Numbers 
    cout<<"Maximum Element From Entered N Numbers is : "<<iret;

    //Deallocating Memory of Array 
    free(obj.arr);

    return 0;
}
